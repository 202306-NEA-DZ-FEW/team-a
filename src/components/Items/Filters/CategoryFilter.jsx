import { Menu, Transition } from "@headlessui/react";
import Link from "next/link";
import { Fragment } from "react";
import { BiCategory } from "react-icons/bi";
import { IoMdArrowDropdown } from "react-icons/io";

import getAllCategories from "@/lib/getAllCategories";

function CategoryFilter({ t, queryParams }) {
  function classNames(...classes) {
    return classes.filter(Boolean).join(" ");
  }
  const categories = getAllCategories(t);

  return (
    <>
      <Menu as='div' className='relative w-full flex justify-center text-left'>
        <Menu.Button className='flex items-center gap-2 w-full justify-between bg-base-200 input-sm rounded-full tracking-wider'>
          <div className='flex gap-2 items-center'>
            <BiCategory className='w-5 h-5' />
            <span>
              {queryParams.category
                ? t(`categories:${queryParams.category}`)
                : t("itemsPage:all")}
            </span>
          </div>
          <IoMdArrowDropdown />
        </Menu.Button>

        <Transition
          as={Fragment}
          enter='transition ease-out duration-100'
          enterFrom='transform opacity-0 scale-95'
          enterTo='transform opacity-100 scale-100'
          leave='transition ease-in duration-75'
          leaveFrom='transform opacity-100 scale-100'
          leaveTo='transform opacity-0 scale-95'
        >
          <Menu.Items className='absolute max-h-52 overflow-y-scroll no-scrollbar bg-base-100 z-10 rounded-lg p-2 top-10 w-full ring-1 ring-black ring-opacity-5 focus:outline-none drop-shadow-xl'>
            <div className='py-1'>
              {categories.map(({ name, dataKey }) => (
                <Link
                  scroll={false}
                  key={dataKey}
                  href={{
                    pathname: "/items",
                    query: { ...queryParams, category: dataKey },
                  }}
                >
                  <Menu.Item>
                    {({ active }) => (
                      <p
                        className={classNames(
                          active ? "bg-base-200" : "text-gray-500",
                          "block px-4 py-2 tracking-wider my-2 rounded-md"
                        )}
                      >
                        {name}
                      </p>
                    )}
                  </Menu.Item>
                </Link>
              ))}
            </div>
          </Menu.Items>
        </Transition>
      </Menu>
    </>
  );
}

export default CategoryFilter;
