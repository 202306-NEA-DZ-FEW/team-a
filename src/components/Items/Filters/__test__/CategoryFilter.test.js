import renderer from "react-test-renderer";

import CategoryFilter from "../CategoryFilter";

it("renders correctly", () => {
  const queryParams = {};
  const mockT = jest.fn();
  const tree = renderer
    .create(<CategoryFilter queryParams={queryParams} t={mockT} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
