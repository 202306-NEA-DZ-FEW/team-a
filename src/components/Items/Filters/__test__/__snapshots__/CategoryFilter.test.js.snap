// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="relative w-full flex justify-center text-left"
  data-headlessui-state=""
>
  <button
    aria-expanded={false}
    aria-haspopup="menu"
    className="flex items-center gap-2 w-full justify-between bg-base-300 input-sm rounded-full tracking-wider"
    data-headlessui-state=""
    id="headlessui-menu-button-:r0:"
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    type="button"
  >
    <div
      className="flex gap-2 items-center"
    >
      <svg
        className="w-5 h-5"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 3H4a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4a1 1 0 0 0-1-1zM9 9H5V5h4v4zm11-6h-6a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1V4a1 1 0 0 0-1-1zm-1 6h-4V5h4v4zm-9 4H4a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h6a1 1 0 0 0 1-1v-6a1 1 0 0 0-1-1zm-1 6H5v-4h4v4zm8-6c-2.206 0-4 1.794-4 4s1.794 4 4 4 4-1.794 4-4-1.794-4-4-4zm0 6c-1.103 0-2-.897-2-2s.897-2 2-2 2 .897 2 2-.897 2-2 2z"
        />
      </svg>
      <span />
    </div>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M128 192l128 128 128-128z"
      />
    </svg>
  </button>
</div>
`;
